lib/travis/api/app/endpoint/builds.rb

```diff
@@ -46,7 +48,17 @@ class Builds < Endpoint
 
       post '/:id/restart' do
         Metriks.meter("api.request.restart_build").mark
-        respond_with service(:reset_model, build_id: params[:id])
+
+        service = self.service(:reset_model, build_id: params[:id])
+        if !service.accept?
+          status 400
+          result = false
+        else
+          Travis::Sidekiq::BuildRestart.perform_async(id: params[:id], user_id: current_user.id)
+          status 200
```

Comment:
I don't think the status needs to be set to 200 explicitly, that's the default. Also, 202 might be a better status here.